src/cljdoc/server/api.clj

```diff
@@ -34,7 +34,8 @@
         (build-log/analysis-received! build-tracker build-id file-uri)
         (ingest/ingest-cljdoc-edn storage data)
         (build-log/api-imported! build-tracker build-id ns-count)
-        (build-log/completed! build-tracker build-id))
+        (build-log/completed! build-tracker build-id)
+        (when (zero? ns-count) (throw (Exception. "No namespaces found"))))
```

Comment:
I'm wondering if maybe `build-log/completed!` shouldn't be called when there's no namespaces. Maybe instead we just call `build-log/failed!` with the "No namespaces found" message?

Reply:
This is the current master behaviour:
![Screenshot 2020-04-21 at 19 43 08](https://user-images.githubusercontent.com/26024693/79896687-f7c97000-8408-11ea-95ab-95f91dcdff47.png)

Here how this commit currently shows it:
![Screenshot 2020-04-21 at 19 51 43](https://user-images.githubusercontent.com/26024693/79897161-95bd3a80-8409-11ea-96bd-0862bcba93fb.png)

Here the result when I implement your suggestion (i.e. do not call completed!):
![Screenshot 2020-04-21 at 20 09 41](https://user-images.githubusercontent.com/26024693/79899176-baff7800-840c-11ea-8a7a-c08c4dd34bee.png)

Also possible is this (does neither call api-imported! nor completed!):
![Screenshot 2020-04-21 at 20 17 01](https://user-images.githubusercontent.com/26024693/79899677-67415e80-840d-11ea-912b-f9a9b1e0b4f5.png)

I do not like the last version, not calling api-imported! seems odd to me. The current PR is the most prudent and least code-intrusive as everything that was called is still called and only a throw of the exception is added for the logging purposes.

Reply:
Ok sounds good! Thanks for taking the time to walk me through it, appreciate that!

Reply:
My pleasure, when the workpiece is already clamped, variations are easy :-)

Reply:
https://youtu.be/ShEez0JkOFw :)